### ISSUE TITLE
[Bugfix] Test and fix the building's creation speed

### ISSUE THREAD

**Issue by joanbm** [jgirald/ES2015C#443]
It has been observed that buildings tend to build very fast, almost instantaneously, which is not intended. It must be checked if this behavior actually occurs, if there is some specific sequence of actions to trigger this behavior, and if it is determined that it occurs, it must be then fixed in the code.

**Product backlog item:** As a player, I want to create buildings by collected resources, so that I can grow my civilization. All buildings are characterized by a construction time and a degree of preservation (similar to degree of life in units). [High priority]

**Estimated effort:** 2h

**Ending condition:** The buildings must build at a moderate speed, slowly incrementing their health from 0 to their maximum HP during a perceptible amount of time. While the building is being created, the partially build model must be used, and when it has been successfully built, the regular model must be used.

**Comment by joanbm** [comment]
I tested the behaviour explained in the issue. It turns out that the feature did work, but the default speed was set way too high, so perceptually, buildings constructed almost immediately.

However, I detected two points where the behaviour could be easily improved:

* The building code was in the Update() call (called once per frame), and didn't adjust for frame time. This means that the building speed per unit of time could vary according to the system, depending on how much it is lagging.

* The building time was always proportional to the number of maximum hit points. This means that it wasn't possible to adjust the building time without modifying the maximum hit points.

To fix those issues, I've modified the code to be independent of the frame time, and added two new properties:

* A building now has a (default) building time, measured in seconds. Therefore, it can be intuitively adjusted on a per-building basis.

* A object that can build (a civil unit, for now) now has a "building factor", which is how fast the unit can build the building relative to the default building time. So if this factor is 2, the building is completed in half the default building time.
